sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/batch/BulkExecutorTest.java

```diff
@@ -0,0 +1,354 @@
+            assertThat(cosmosBulkItemResponse.getSessionToken()).isNotNull();
+            assertThat(cosmosBulkItemResponse.getActivityId()).isNotNull();
+            assertThat(cosmosBulkItemResponse.getRequestCharge()).isNotNull();
+        }
+
+        int iterations = 0;
+        while (true) {
+            assertThat(iterations < 100);
+            if (executor.isDisposed()) {
+                break;
+            }
+
+            Thread.sleep(10);
+            iterations++;
+        }
+    }
+
+    static protected void safeClose(CosmosAsyncClient client) {
+        if (client != null) {
+            try {
+                client.close();
+            } catch (Exception e) {
+                logger.error("failed to close client", e);
+            }
+        }
+    }
+
+    static protected void safeCloseAsync(CosmosAsyncClient client) {
+        if (client != null) {
+            new Thread(() -> {
+                try {
+                    client.close();
+                } catch (Exception e) {
+                    logger.error("failed to close client", e);
+                }
+            }).start();
+        }
+    }
+
+    static protected void safeCloseClient(CosmosAsyncClient client) {
+        if (client != null) {
+            try {
+                logger.info("closing client ...");
+                client.close();
+                logger.info("closing client completed");
+            } catch (Exception e) {
+                logger.error("failed to close client", e);
+            }
+        }
+    }
+
+    static protected void safeDeleteAllCollections(CosmosAsyncDatabase database) {
+        if (database != null) {
+            List<CosmosContainerProperties> collections = database.readAllContainers()
+                                                                  .collectList()
+                                                                  .block();
+
+            for (CosmosContainerProperties collection : collections) {
+                database.getContainer(collection.getId()).delete().block();
+            }
+        }
+    }
+
+    static protected void safeDeleteCollection(CosmosAsyncContainer collection) {
+        if (collection != null) {
+            try {
+                collection.delete().block();
+            } catch (Exception e) {
+            }
+        }
+    }
+
+    static protected void safeDeleteCollection(CosmosAsyncDatabase database, String collectionId) {
+        if (database != null && collectionId != null) {
+            try {
+                database.getContainer(collectionId).delete().block();
+            } catch (Exception e) {
+            }
+        }
+    }
+
+    static protected void safeDeleteDatabase(CosmosAsyncDatabase database) {
+        if (database != null) {
+            try {
+                database.delete().block();
+            } catch (Exception e) {
+            }
+        }
+    }
+
+    static protected void safeDeleteSyncDatabase(CosmosDatabase database) {
```

Comment:
nit: this method also does not seem to be used in this test

Reply:
Fixed